### PR TITLE
several cleanups in template type checker

### PR DESCRIPTION
**Pull request by JoostK**
Review per commit.

### CODE REVIEW

File: packages/compiler-cli/src/ngtsc/typecheck/src/source.ts [hunk 1]
```diff
@@ -44,35 +44,35 @@ export class TemplateSource {
...
    * attached to a TCB's function declaration as leading trivia. This enables translation of
    * diagnostics produced for TCB code to their source location in the template.
    */
-  private templateSources = new Map<string, TemplateSource>();
+  private templateSources = new Map<TemplateId, TemplateSource>();
 
-  captureSource(mapping: TemplateSourceMapping, file: ParseSourceFile): string {
-    const id = `tcb${this.nextTcbId++}`;
+  captureSource(mapping: TemplateSourceMapping, file: ParseSourceFile): TemplateId {
+    const id = `tcb${this.nextTemplateId++}` as unknown as TemplateId;
```
Comment (alxhub) [inline packages/compiler-cli/src/ngtsc/typecheck/src/source.ts]: Just a nit: the `as unknown` intermediate cast is unnecessary as `TemplateId` is already a subtype of `string`, so a direct cast is legal.

**Review by alxhub — APPROVED** [review]
👍 nice work! The deduplication of error messages is a huge dx improvement.

**Comment by alxhub** [comment]
[Presubmit](https://test.corp.google.com/ui#id=OCL:286018010:BASE:286018109:1576609901981:a51a0e23)
[Ivy Presubmit](https://test.corp.google.com/ui#id=OCL:286018010:BASE:286018109:1576609901991:9c3c19f)

**Comment by JoostK** [comment]
merge-assistance: presubmit results should be available in Alex's run yesterday.

**Pull request closed without merging**